dashboard/app/controllers/openai_chat_controller.rb

```diff
@@ -8,7 +8,17 @@ def chat_completion
       return render(status: :bad_request, json: {})
     end
 
-    messages = params[:messages]
+    if has_level_id_param?
+      level_id = params[:levelId]
+      level = Level.find(level_id)
+      test_file = level.validation["PainterPlusTest.java"]["text"]
```

Comment:
- [ ] we need to have a dynamic file name here rather than a hardcoded one; figure out where to find that 🤔